setup-local/src/binaryControl.js

```diff
@@ -168,26 +194,39 @@ class BinaryControl {
    * Starts Local Binary using the args generated for this action
    */
   async startBinary() {
-    try {
-      this._generateArgsForBinary();
-      let { localIdentifier } = this.stateForBinary;
-      localIdentifier = localIdentifier ? `with local-identifier=${localIdentifier}` : '';
-      core.info(`Starting local tunnel ${localIdentifier} in daemon mode...`);
+    let { localIdentifier } = this.stateForBinary;
+    localIdentifier = localIdentifier ? `with local-identifier=${localIdentifier}` : '';
+    core.info(`Starting local tunnel ${localIdentifier} in daemon mode...`);
 
-      const { output, error } = await this._triggerBinary(LOCAL_TESTING.START);
+    let triesAvailable = BINARY_MAX_TRIES;
+
+    while (triesAvailable--) {
+      try {
+        this._generateArgsForBinary();
+
+        // eslint-disable-next-line no-await-in-loop
+        const { output, error } = await this._triggerBinary(LOCAL_TESTING.START);
+
+        if (!error) {
+          const outputParsed = JSON.parse(output);
+          if (outputParsed.state === LOCAL_BINARY_TRIGGER.START.CONNECTED) {
+            core.info(`Local tunnel status: ${outputParsed.message}`);
+            return;
+          }
 
-      if (!error) {
-        const outputParsed = JSON.parse(output);
-        if (outputParsed.state === LOCAL_BINARY_TRIGGER.START.CONNECTED) {
-          core.info(`Local tunnel status: ${outputParsed.message}`);
-        } else {
           throw Error(JSON.stringify(outputParsed.message));
+        } else {
+          throw Error(JSON.stringify(error));
+        }
+      } catch (e) {
```

Comment:
is it possible to narrow down the exception type? would be better if we add retry only for specific exceptions and not on generalised.

Reply:
Well... types in JS. Will have to go with TypeScript then.
And regarding re-trying it for specific errors... I don't really have the exhaustive list of when binary might fail to start. But whatever be the error... we can retry it before we throw errors to users CI workflow. As in.. missing a valid error case for retry will give out false positives/negatives...

Reply:
one valid case is obviously not to retry if the access key is invalid. But that's more of a good-to-have feature and can be added later if required...

Reply:
resolving it